src/lib.rs

```diff
@@ -312,6 +317,60 @@ mod contract {
         );
     }
 
+    /// Allow receiving NEP141 tokens to the EVM contract.
+    ///
+    /// This function returns the amount of tokens to return to the sender.
+    /// Either all tokens are transferred tokens are returned in case of an
+    /// error, or no token is returned if tx was successful.
+    #[no_mangle]
+    pub extern "C" fn ft_on_transfer() {
+        let mut engine = Engine::new(predecessor_address()).sdk_unwrap();
+
+        let args: NEP141FtOnTransferArgs = parse_json(sdk::read_input().as_slice())
+            .sdk_unwrap()
+            .try_into()
+            .map_err(|err| format!("ERR_JSON_{:?}", err))
+            .sdk_unwrap();
+
+        if sdk::predecessor_account_id() == sdk::current_account_id() {
+            let engine = Engine::new(predecessor_address()).sdk_unwrap();
+            EthConnectorContract::get_instance().ft_on_transfer(&engine, &args);
+        } else {
+            engine.receive_erc20_tokens(&args);
+        }
+    }
+
+    /// Deploy ERC20 token mapped to a NEP141
+    #[no_mangle]
+    pub extern "C" fn deploy_erc20_token() {
+        // Id of the NEP141 token in Near
+        let args: DeployErc20TokenArgs =
+            DeployErc20TokenArgs::try_from_slice(&sdk::read_input()).sdk_expect("ERR_ARG_PARSE");
+
+        let mut engine = Engine::new(predecessor_address()).sdk_unwrap();
+
+        let erc20_contract = include_bytes!("../etc/eth-contracts/res/EvmErc20.bin");
+        let deploy_args = ethabi::encode(&[
+            ethabi::Token::String("Empty".to_string()),
+            ethabi::Token::String("EMPTY".to_string()),
+            ethabi::Token::Uint(ethabi::Uint::from(0)),
```

Comment:
How these values are expected to be updated later?

Reply:
For now using admin privileges! Later we should create a trustless way to bridge metadata.

Reply:
What's the interface for upgrading the `name` and `symbol`?
Are we going to use `adminSstore` each time to change metadata until we create a new trustless way?

Reply:
yes, for now admin Sstore